integrations/terraform-cloud/client.py

```diff
@@ -0,0 +1,174 @@
+import httpx
+from typing import Any, AsyncGenerator, Optional
+from loguru import logger
+from port_ocean.context.event import event
+
+TERRAFORM_WEBHOOK_EVENTS = [
+    "run:applying",
+    "run:completed",
+    "run:created",
+    "run:errored",
+    "run:needs_attention",
+    "run:planning",
+]
+
+PAGE_SIZE = 100
+
+class TerraformClient:
+    def __init__(self, terraform_base_url: str, auth_token: str) -> None:
+        self.terraform_base_url = terraform_base_url
+        self.base_headers = {
+            "Authorization": f"Bearer {auth_token}",
+            "Content-Type": "application/vnd.api+json",
+        }
+        self.api_url = f"{self.terraform_base_url}/api/v2"
+        self.client = httpx.AsyncClient(headers=self.base_headers)
+
+    async def send_api_request(
+        self,
+        endpoint: str,
+        method: str = "GET",
+        query_params: Optional[dict[str, Any]] = None,
+        json_data: Optional[dict[str, Any]] = None,
+    ) -> dict[str, Any]:
+        logger.info(f"Requesting Terraform Cloud data for endpoint: {endpoint}")
+        try:
+            url = f"{self.api_url}/{endpoint}"
+            logger.debug(f"URL: {url}, Method: {method}, Params: {query_params}, Body: {json_data}")
+            response = await self.client.request(
+                method=method,
+                url=url,
+                params=query_params,
+                json=json_data,
+            )
+            response.raise_for_status()
+
+            logger.info(f"Successfully retrieved data for endpoint: {endpoint}")
+            logger.debug(f"Response: {response.json()}")
+
+            return response.json()
+
+        except httpx.HTTPStatusError as e:
+            logger.error(f"HTTP error on {endpoint}: {e.response.status_code} - {e.response.text}")
+            raise
+        except httpx.HTTPError as e:
+            logger.error(f"HTTP error on {endpoint}: {str(e)}")
+            raise
+
+    async def get_paginated_resources(
+        self, endpoint: str, params: Optional[dict[str, Any]] = None
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        params = params or {}
+        params.update({"page[size]": PAGE_SIZE})
+        next_url = endpoint
+
+        while next_url:
+            response = await self.send_api_request(endpoint=next_url, query_params=params)
+            resources = response.get("data", [])
+
+            logger.info(f"Fetched {len(resources)} resources from {next_url}")
+            yield resources
+
+            next_url = response.get("links", {}).get("next", "")
+            if not next_url:
+                logger.debug(f"No more pages to fetch for {endpoint}")

```

Comment:
I am not sure of the exact behavior of `requests` in this situation but please verify it.

the next link contains params included in the URL 
e.g. `"next": "https://app.terraform.io/api/v2/organizations/hashicorp/workspaces?page%5Bnumber%5D=2&page%5Bsize%5D=20",`

make sure that when you pass yours params after the first fetch it doesn't override the params embedded in the URL but rather appends to it.